### PR TITLE
sys/usb_board_reset: add usb_board_reset_in_bootloader definition

### PR DESCRIPTION
**Pull request by benpicco**
Move the definition of `usb_board_reset_in_bootloader()` to a common location.
Inclusion of `usb_board_reset.h` needs to be guarded with `#define USB_H_USER_IS_RIOT_INTERNAL`.
We could move that definition into `usb_board_reset.h` instead.

### Contribution description

Code should keep compiling.

### Issues/PRs references

suggested in https://github.com/RIOT-OS/RIOT/pull/14378#issuecomment-652008940

### CODE REVIEW

File: sys/shell/commands/sc_sys.c [hunk 1]
```diff
@@ -22,6 +22,9 @@
 
 #include "periph/pm.h"
 
+#define USB_H_USER_IS_RIOT_INTERNAL
+#include "usb_board_reset.h"
```
Comment (aabadie) [inline sys/shell/commands/sc_sys.c:26]:
```suggestion
#ifdef MODULE_USB_BOARD_RESET
#define USB_H_USER_IS_RIOT_INTERNAL
#include "usb_board_reset.h"
#endif
```

Comment (benpicco) [reply sys/shell/commands/sc_sys.c:27]: The include is safe to use either way, but added.

**Comment by benpicco** [comment]
@chrysn if you want you can just push to this branch.

**Comment by chrysn** [comment]
Thanks, pushed. I'm not all sure how reviewing PRs involving own code works, but ACK from me on the first commit FWIW.

File: boards/adafruit-clue/reset.c [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 #define USB_H_USER_IS_RIOT_INTERNAL
 
+#include "cpu.h"
```
Comment (aabadie) [inline boards/adafruit-clue/reset.c:24]: Why is this needed now ?

Comment (benpicco) [reply boards/adafruit-clue/reset.c:24]: The definition for the `NRF_POWER` register was previously pulled in through `usb_board_reset.h` -> `usb/usbus/cdc/acm.h` -> `usb.h` -> `board.h` -> `cpu.h`

Comment (aabadie) [reply boards/adafruit-clue/reset.c:24]: Ok, I see. Makes sense.
